Diagnostic/Utils/LadDiagnosticUtil.py

```diff
@@ -85,47 +94,50 @@ def generateOMIQueries(omiQueryConfiguration):
-        tableName = performanceCounter['table']
-        if 'condition' in performanceCounter:
-            condition = performanceCounter['condition']
+        class_name = getOmiClassName(performance_counter['class'])
+        table_name = performance_counter['table']
+        if 'condition' in performance_counter:
+            condition = performance_counter['condition']
         else:
             condition = None
-        if 'namespace' in performanceCounter:
-            namespace = performanceCounter['namespace']
+        if 'namespace' in performance_counter:
+            namespace = performance_counter['namespace']
         else:
             namespace = 'root/scx'
-        if not tableName in omiQueryConfiguration:
-            omiQueryConfiguration[tableName] = defaultdict(defaultdict)
-        if not namespace in omiQueryConfiguration[tableName]:
-            omiQueryConfiguration[tableName][namespace] = defaultdict(queryParameter)
-        if not className in omiQueryConfiguration[tableName][namespace]:
-            omiQueryConfiguration[tableName][namespace][className] = queryParameter()
-            omiQueryConfiguration[tableName][namespace][className].queryCondition = condition
+        if not table_name in omi_queries:
+            omi_queries[table_name] = defaultdict(defaultdict)
+        if not namespace in omi_queries[table_name]:
+            omi_queries[table_name][namespace] = defaultdict(QueryParameter)
+        if not class_name in omi_queries[table_name][namespace]:
+            omi_queries[table_name][namespace][class_name] = QueryParameter()
+            omi_queries[table_name][namespace][class_name].queryCondition = condition
         else:
-            if omiQueryConfiguration[tableName][namespace][className].queryCondition != condition:
+            if omi_queries[table_name][namespace][class_name].queryCondition != condition:
                 raise Exception('Cannot have two different conditions on same table')
-        if not performanceCounter['counterSpecifier'] in omiQueryConfiguration[tableName][namespace][className].querySelectParameters:
-            omiQueryConfiguration[tableName][namespace][className].append(performanceCounter['counterSpecifier'])
-    return omiQueryConfiguration
+        if not performance_counter['counterSpecifier'] in omi_queries[table_name][namespace][class_name].querySelectParameters:
+            omi_queries[table_name][namespace][class_name].append(performance_counter['counterSpecifier'])
+    return omi_queries
+
 
 # Get elements from DiagnosticsMonitorConfiguration in LadCfg based on element name
 def getDiagnosticsMonitorConfigurationElement(ladCfg, elementName):
-    if ladCfg:
-        if 'diagnosticMonitorConfiguration' in ladCfg:
-            if elementName in ladCfg['diagnosticMonitorConfiguration']:
-                return ladCfg['diagnosticMonitorConfiguration'][elementName]
+    if ladCfg and 'diagnosticMonitorConfiguration' in ladCfg:
```

Comment:
This style of data retrieval should be hoisted to a more general function. The names should probably be turned into constants, but the names are themselves part of the negotiation configuration protocol between the Portal diagnostics config blade and the agent. Pascal casing is the norm, but sometimes mistakes get through and you just have to deal with them.